app/tracing.ts

```diff
@@ -0,0 +1,227 @@
+import { BatchSpanProcessor, SamplingDecision, SamplingResult } from '@opentelemetry/sdk-trace-base';
+import { Resource } from '@opentelemetry/resources';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { AWSXRayIdGenerator } from '@opentelemetry/id-generator-aws-xray';
+import { Attributes, Context, Link, SpanKind, trace } from '@opentelemetry/api';
+import { AWSXRayPropagator } from '@opentelemetry/propagator-aws-xray';
+import env from './util/env';
+import { random } from 'lodash';
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk';
+
+
+
+// Register instrumentation libraries for Node.js auto-instrumentation and capturing of
+// outgoing HTTP calls
+registerInstrumentations({
+  instrumentations: [
+    getNodeAutoInstrumentations(),
+    new HttpInstrumentation(),
+    new AwsInstrumentation({}),
+  ],
+});
+
+const resource =
+  Resource.default().merge(
+    new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: env.harmonyClientId,
+      [SemanticResourceAttributes.SERVICE_VERSION]: '0.1.0',
+    }),
+  );
+interface SampleCheckContext {
+  context?: Context,
+  traceId?: string,
+  spanName?: string,
+  spanKind?: SpanKind,
+  attributes?: Attributes,
+  links?: Link[],
+}
+
+// maximum number of rejected traces we will track to avoid sending partial traces
+const REJECTED_TRACE_CACHE_LIMIT = 10000;
+
+// class used to track to the traces we have rejected so we don't send subsegments that have no
+// parent (due to the parent being rejected).
+// TODO This should probably be tracked externally in
```

Comment:
Usually when we have a TODO we have a ticket number. I'm not sure if you think we should open a JIRA ticket.

Reply:
Yeah - I thought about that, but then I thought we should try things out for a while and see if it is really necessary.